Added information regarding failed email delivery to UPGRADING.md

hyfloac opened the pull request:
Several people have been experiencing issues with emails getting stuck in queue since the v0.12.0 update. I personally experienced this issue, and I've seen at least 2 other people on the Discord who were having similar issues.

Based on my testing, and the discussion at https://github.com/stalwartlabs/stalwart/discussions/1579, this change tries to document some of the possible ways to resolve this issue.

The solutions proposed in the UPGRADING.md document now include attempting to fix ones DNS resolver to pass DNSSEC options, attempting to disable `resolver.edns` (which did not work for me, but I'll leave as a suggestion for others to attempt), and disabling DANE checks on outbound emails.

entelligence-ai-reviews commented:
## Walkthrough
This update enhances the upgrade documentation by adding a new section that addresses a known mail delivery issue introduced in version 0.12.0. The section details problems related to EDNS resolution being enabled by default for DANE support, outlines potential causes, and provides troubleshooting steps for users encountering delivery failures after upgrading. No existing documentation was altered; the change is purely additive to assist with post-upgrade issues.

## Changes
| File(s)      | Summary |
| ---          | --- |
| UPGRADING.md | Added a new 'Issues with Delivery' section describing a known EDNS/DANE-related mail delivery issue in v0.12.0, with troubleshooting steps. |

## Sequence Diagram
This diagram shows the interactions between components:

```mermaid
sequenceDiagram
    title Stalwart Mail Delivery Process with DANE/EDNS Issues

    participant User as "User"
    participant Stalwart as "Stalwart Mail Server"
    participant DNS as "DNS Resolver"
    participant Remote as "Remote Mail Server"
    
    User->>Stalwart: Send email
    activate Stalwart
    
    Note over Stalwart: v0.12.0 enabled EDNS by default for DANE support
    
    Stalwart->>DNS: Query TLSA records<br/>(_25._tcp.domain TLSA +dnssec)
    activate DNS
    
    alt DNS resolver returns EDNS information
        DNS-->>Stalwart: Return TLSA and RRSIG records
        Stalwart->>Remote: Establish TLS connection with DANE verification
        Remote-->>Stalwart: TLS connection established
        Stalwart->>Remote: Deliver email
        Remote-->>Stalwart: Delivery successful
        Stalwart-->>User: Email sent successfully
    
    else DNS resolver fails to return EDNS information
        DNS-->>Stalwart: Incomplete response (missing RRSIG)
        Note over Stalwart: Mail delivery fails
        
        alt Solution 1: Fix DNS resolver
            Note over User, DNS: Ensure resolver returns EDNS information
        
        else Solution 2: Change DNS resolver
            User->>Stalwart: Configure different DNS resolver
            Note over Stalwart: Settings -> SMTP -> Outbound -> DNS -> Resolver
        
        else Solution 3: Disable EDNS
            User->>Stalwart: Set resolver.edns to false in config
        
        else Solution 4: Disable DANE
            User->>Stalwart: Set DANE to "disabled"
            Note over Stalwart: Settings -> SMTP -> Outbound -> TLS -> DANE
        end
        
        Stalwart->>Remote: Deliver email (with adjusted settings)
        Remote-->>Stalwart: Delivery successful
        Stalwart-->>User: Email sent successfully
    end
    
    deactivate DNS
    deactivate Stalwart
```


▶️ ⚡ **AI Code Reviews for VS Code, Cursor, Windsurf**  
[Install the extension](https://marketplace.visualstudio.com/items?itemName=EntelligenceAI.EntelligenceAI-PR-Reviewer)

<details>
<summary>Note for Windsurf</summary>
Please change the default marketplace provider to the following in the windsurf settings:

Marketplace Extension Gallery Service URL: https://marketplace.visualstudio.com/_apis/public/gallery

Marketplace Gallery Item URL: https://marketplace.visualstudio.com/items
</details>

<details>
<summary>Entelligence.ai can learn from your feedback. Simply add 👍 / 👎 emojis to teach it your preferences. More shortcuts below</summary>

### **Emoji Descriptions:**
- ⚠️ **Potential Issue** - May require further investigation.
- 🔒 **Security Vulnerability** - Fix to ensure system safety.
- 💻 **Code Improvement** - Suggestions to enhance code quality.
- 🔨 **Refactor Suggestion** - Recommendations for restructuring code.
- ℹ️ **Others** - General comments and information.

### **Interact with the Bot:**
- Send a message or request using the format:
`@entelligenceai + *your message*`
```plaintext
Example: @entelligenceai Can you suggest improvements for this code?
```
- Help the Bot learn by providing feedback on its responses.
`@entelligenceai + *feedback*`
```plaintext
Example: @entelligenceai Do not comment on `save_auth` function !
```

Also you can trigger various commands with the bot by doing
`@entelligenceai command`

The current supported commands are
1. `config` - shows the current config
2. `retrigger_review` - retriggers the review

More commands to be added soon.
</details>

stappersg commented:
IMNSHO is the merge request way too good.  I plea for moving _"Issues with Delivery"_ outside `UPGRADING.md`, to a more generic place.

hyfloac committed:
Added upgrade information regarding failed email delivery

Several people, including me, have been experiencing issues with emails getting stuck in Queue since the v0.12.0 update.

Based on my testing, and the discussion at #1579, this change tries to document some of the possible ways to resolve this issue.

mdecimus commented:
Thanks but this change is no longer necessary, v0.12.4 has the hickory update that fixes this.
